homeworks/week02/README.md

```diff
@@ -0,0 +1,232 @@
+receive this in the form of an array of simple JavaScript objects,
+which your code will convert into HTML. You will also add some simple
+CSS to style the page.
+
+To begin, clone this repo to your local machine. You should be able to
+open the ```index.html``` file in your browser and see a simple
+```Hello, world.``` message.
+
+Open and examine the ```index.html``` file. You will see a function with the following signature:
+
+```javascript
+function render(movieArray, displayWide)
+```
+
+This function will be automagically called for you when you open or
+reload this webpage, so you can think of this function as the "entry
+point" to the code that you will write for this assignment. This
+function must return a string, and the string that it returns will get 
+automagically inserted into the ```<body>``` of the HTML page for you.
+
+ The function has two arguments: ```movieArray``` will contain an
+ array of "movie" objects, which we will explain below, and
+ ```displayWide``` is a boolean which (in [Step 4](#user-content-step-4)) 
+ will control what kind of HTML you should generate, but you can ignore
+ it for now.
+
+## Step 1:
+
+Begin by creating a function that implements the following signature:
+```javascript
+function renderMovie(movie)
+```
+
+The ```movie``` argument will be a JavaScript object with five fields,
+like this example:
+
+```javascript
+{
+	name: "Minions",
+	year: 2015,
+	genre: "animation",
+	cast: ["Sandra Bullock", "Jon Hamm", "Michael Keaton"],
+	description: "Evolving from single-celled yellow organisms at the dawn of time, Minions live to serve, but find themselves working for a continual series of unsuccessful masters, from T. Rex to Napoleon. Without a master to grovel for, the Minions fall into a deep depression. But one minion, Kevin, has a plan."
+}
+```
+
+Using the attributes of this object, your function should create a new
+string variable, use string concatenation to generate a string of
+HTML, and return that variable with the ```return``` command.
+
+For example, given the above object, you should generate the following
+HTML:
+
+```html
+<div class="movie animation">
+  <h1 class="name">Minions</h1>
+  <p>Release date: <span class="year">2015</span>, Genre: <span class="genre">animation</span>.</p>
+
+  <div>
+    Starring:
+    <ul>
+      <li>Sandra Bullock</li>
+      <li>Jon Hamm</li>
+      <li>Michael Keaton</li>
+    </ul>
+  </div>
+  <p class="description">
+    Evolving from single-celled yellow organisms at the dawn of time,
+    Minions live to serve, but find themselves working for a continual
+    series of unsuccessful masters, from T. Rex to Napoleon. Without a
+    master to grovel for, the Minions fall into a deep depression. But
+    one minion, Kevin, has a plan.
+  </p>
+</div>
+```
+
+For example, to get you started, you might begin with something like
+the following:
+```javascript
+  var returnValue = "";
+
+  returnValue += "<div class=\"movie " + movie.genre + "\">";
```

Comment:
Use single quotes for js string literals, because that's our coding style, and also it saves you the trouble of escaping double quotes.